### ISSUE TITLE
sourcemap vs sourceMap inconsistencies

### ISSUE THREAD

**Issue by staxmanade** [Microsoft/TypeScript#3118]
Hello,

Wanted to highlight that there appears to be a difference between the use of `sourceMap` vs `sourcemap` within the project (likely just documentation related).

While using the [gulp-tsc](https://www.npmjs.com/package/gulp-tsc) plugin I found it difficult to figure out what `options` could be passed through to the typescript compiler. One way I could figure out what options could be passed was to use the `tsc --help` command line command and would print the following options:

```
Version 1.5.0-beta
Syntax:   tsc [options] [file ...]

Examples: tsc hello.ts
          tsc --out file.js file.ts
          tsc @args.txt

Options:
 -d, --declaration                  Generates corresponding '.d.ts' file.
 -h, --help                         Print this message.
 --mapRoot LOCATION                 Specifies the location where debugger should locate map files instead of generated locations.
 -m KIND, --module KIND             Specify module code generation: 'commonjs' or 'amd'
 --noEmit                           Do not emit outputs.
 --noEmitOnError                    Do not emit outputs if any type checking errors were reported.
 --noImplicitAny                    Raise error on expressions and declarations with an implied 'any' type.
 --out FILE                         Concatenate and emit output to single file.
 --outDir DIRECTORY                 Redirect output structure to the directory.
 --preserveConstEnums               Do not erase const enum declarations in generated code.
 -p DIRECTORY, --project DIRECTORY  Compile the project in the given directory.
 --removeComments                   Do not emit comments to output.
 --rootDir LOCATION                 Specifies the root directory of input files. Use to control the output directory structure with --outDir.
 --sourceMap                        Generates corresponding '.map' file.
 --sourceRoot LOCATION              Specifies the location where debugger should locate TypeScript files instead of source locations.
 --suppressImplicitAnyIndexErrors   Suppress noImplicitAny errors for indexing objects lacking index signatures.
 -t VERSION, --target VERSION       Specify ECMAScript target version: 'ES3' (default), 'ES5', or 'ES6' (experimental)
 -v, --version                      Print the compiler's version.
 -w, --watch                        Watch input files.
 @<file>                            Insert command line options and files from a file.
```

Notice how the ` --sourceMap` is specified above, but when I passed `{sourceMap: true}` to the `gulp-tsc` it did not generate the sourcemaps. I tried toggling the `--sourceRoot...` option and it gave me an error saying that the `sourcemap` property had to be on (which was a clue as to why my sourcemaps were not being generated).

Any thoughts about updating everywhere in the project where `sourceMap` is used to include be `sourcemap` instead?

**Comment by DanielRosenwasser** [comment]
Our compiler will accept `sourceMap`, `sourcemap`, or even `sOuRcEmAp` if that's your kind of thing. This sounds like an issue with `gulp-tsc`.

**Comment by DanielRosenwasser** [comment]
But I think I see part of the problem you've found; our error messages actually use the casing `sourcemap` instead of `sourceMap` consistently.

**Comment by staxmanade** [comment]
:+1: I like the `sOuRcEmAp` option :tongue: 

You're right loos like an issue in `gulp-tsc` https://github.com/kant2002/gulp-tsc/blob/master/lib/compiler.js#L38

**Comment by basarat** [comment]
`tsconfig.json` is picky about casing : https://github.com/Microsoft/TypeScript/issues/1933 < just wanted to point that out :rose:

**Comment by DanielRosenwasser** [comment]
Yup, I still think we should be case insensitive, but that's just me.
